web_last_viewed_records/static/src/js/main.js

```diff
@@ -0,0 +1,174 @@
+openerp.web_last_viewed_records = function(instance){
+    var QWeb = instance.web.qweb;
+    var _t = instance.web._t;
+
+    instance.web.ActionManager.include({
+        last_viewed_history_var: 'odoo_last_viewed',
+        last_viewed_history_size: 8,
+        init:function(){
+            this._super.apply(this, arguments);
+
+            this.last_viewed = [];
+            var last_viewed = this.load_last_viewed_history().last_viewed || [];
+            for (var i=last_viewed.length-1; i>=0; i--){
+                this.unshift_last_viewed(last_viewed[i]);
+            }
+        },
+        add_last_viewed: function(item){
+            if (_.any(this.last_viewed, function(x){
+                    if (x['title'] != item['title'] ||
+                        x['view_type'] != item['view_type'] ||
+                        x['url']['model'] != item['url']['model'])
+                        return false;
+                    if (x['view_type'] == 'form' && x['id'] != item['id'])
+                        return false;
+                    return true;
+                }))
+                return;
+
+            //save json data to localStorage
+            var data = this.load_last_viewed_history();
+            data.last_viewed = data.last_viewed || []
+            data.last_viewed.unshift(item);
+            data.last_viewed.splice(this.last_viewed_history_size);
+            this.save_last_viewed_history(data);
+
+            this.unshift_last_viewed(item);
+        },
+        unshift_last_viewed: function(item){
+            var self = this;
+            this.last_viewed.unshift(item);
+            this.last_viewed.splice(this.last_viewed_history_size);
+        },
+        load_last_viewed_history: function(){
+            var data = localStorage[this.last_viewed_history_var] || '{}';
+            return JSON.parse(data);
+        },
+        save_last_viewed_history: function(data){
+            localStorage[this.last_viewed_history_var] = JSON.stringify(data);
+        },
+        // icon map: http://bistro.convergencecms.co/entypo
+        _model2icon: {
+            'res.partner':'+',
+            'crm.lead':'4',
+            'sale.order':'l',
+            'account.analytic.account':'7',
+            'crm.phonecall':'!',
+            'hr.employee':'.',
+            'hr.applicant':'-',
+            'project.project':'t',
+            'project.task':'W',
+            'account.invoice':'h',
+            'ir.module.module':'Z',
+            'hr_timesheet_sheet.sheet': 'N',
+            'res.groups': ',',
+            'res.company': '_',
+            'res.user': 'ó',
+            'gamification.challenge':'è',
+            'gamification.badge':'8',
```

Comment:
Another great option is if you can support Font awesome here: http://fortawesome.github.io/Font-Awesome/icons/ It's already used in Odoo in smart-buttons.